aten/src/ATen/core/SingletonSymNodeImpl.h

```diff
@@ -132,6 +139,18 @@ class C10_API SingletonSymNodeImpl : public SymNodeImpl {
     return coeff_;
   }
 
+  const at::Tensor& singleton_data() {
+    return data_;
+  }
+
+  void* singleton_dummy() override {
```

Comment:
I may be able to return `TensorImpl*` instead of `void*` here since TensorImpl is actually defined in c10.